client/mysqlbinlog.cc

```diff
@@ -3225,6 +3317,8 @@ int main(int argc, char** argv)
   DBUG_RETURN(retval == ERROR_STOP ? 1 : 0);
 
 err:
+  if (progress_file != NULL)
+    my_fclose(progress_file, MYF(0));
```

Comment:
It looks like the close condition may be better in `cleanup()` to handle some error paths.

Reply:
Thank you. Updated in latest commit.